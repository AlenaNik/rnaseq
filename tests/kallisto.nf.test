nextflow_pipeline {

    name "Test pipeline with Kallisto, skipping both QC and alignment"
    script "../main.nf"

    test("Params: --pseudo_aligner kallisto --skip_qc --skip_alignment") {

        when {
            params {
                outdir = "$outputDir"
                pseudo_aligner = 'kallisto'
                skip_qc = true
                skip_alignment = true
            }
        }

        then {
            def path_not_snapshottable = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
            def path_snapshottable     = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    path_not_snapshottable.collect{ file -> path_snapshottable.contains(file) ? null : file.getName()}.findAll{ it },
                    path_snapshottable
                ).match() }
            )
        }
    }

    test("Params: --pseudo_aligner kallisto --skip_qc --skip_alignment - stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
                pseudo_aligner = 'kallisto'
                skip_qc = true
                skip_alignment = true
            }
        }

        then {
            def path_not_snapshottable = UTILS.getAllFilesFromDir(params.outdir, null, UTILS.exclusionRegexesForUnstableFileNames)
            def path_snapshottable     = UTILS.getAllFilesFromDir(params.outdir, UTILS.snapshottablePatterns, UTILS.exclusionRegexesForUnstableFileContents)

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml"),
                    path_not_snapshottable.collect{ file -> path_snapshottable.contains(file) ? null : file.getName()}.findAll{ it },
                    path_snapshottable
                ).match() }
            )
        }
    }
}
