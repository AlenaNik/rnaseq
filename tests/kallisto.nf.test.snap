{
    "Params: --pseudo_aligner kallisto --skip_qc --skip_alignment": {
        "content": [
            {
                "BBMAP_BBSPLIT": {
                    "bbmap": 39.01
                },
                "CAT_FASTQ": {
                    "cat": 8.3
                },
                "CUSTOM_CATADDITIONALFASTA": {
                    "python": "3.9.5"
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "CUSTOM_TX2GENE": {
                    "python": "3.9.5"
                },
                "FQ_SUBSAMPLE": {
                    "fq": "0.9.1 (2022-02-22)"
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "GTF_FILTER": {
                    "python": "3.9.5"
                },
                "GUNZIP_ADDITIONAL_FASTA": {
                    "gunzip": 1.1
                },
                "GUNZIP_GTF": {
                    "gunzip": 1.1
                },
                "KALLISTO_INDEX": {
                    "kallisto": "0.48.0"
                },
                "KALLISTO_QUANT": {
                    "kallisto": "0.48.0"
                },
                "SALMON_QUANT": {
                    "salmon": "1.10.1"
                },
                "SE_GENE": {
                    "bioconductor-summarizedexperiment": "1.32.0"
                },
                "TRIMGALORE": {
                    "trimgalore": "0.6.7",
                    "cutadapt": 3.4
                },
                "TXIMETA_TXIMPORT": {
                    "bioconductor-tximeta": "1.20.1"
                },
                "UNTAR_SALMON_INDEX": {
                    "untar": 1.34
                },
                "Workflow": {
                    "nf-core/rnaseq": "v3.16.0dev"
                }
            },
            [
                "RAP1_IAA_30M_REP1",
                "RAP1_IAA_30M_REP1.stats.txt",
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.html",
                "RAP1_IAA_30M_REP1_trimmed_1_val_1_fastqc.zip",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.html",
                "RAP1_IAA_30M_REP1_trimmed_2_val_2_fastqc.zip",
                "RAP1_UNINDUCED_REP1",
                "RAP1_UNINDUCED_REP1.stats.txt",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.html",
                "RAP1_UNINDUCED_REP1_trimmed_trimmed_fastqc.zip",
                "RAP1_UNINDUCED_REP2",
                "RAP1_UNINDUCED_REP2.stats.txt",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.html",
                "RAP1_UNINDUCED_REP2_trimmed_trimmed_fastqc.zip",
                "WT_REP1",
                "WT_REP1.stats.txt",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_1_val_1_fastqc.html",
                "WT_REP1_trimmed_1_val_1_fastqc.zip",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_2_val_2_fastqc.html",
                "WT_REP1_trimmed_2_val_2_fastqc.zip",
                "WT_REP2",
                "WT_REP2.stats.txt",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_1_val_1_fastqc.html",
                "WT_REP2_trimmed_1_val_1_fastqc.zip",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_2_val_2_fastqc.html",
                "WT_REP2_trimmed_2_val_2_fastqc.zip",
                "abundance.h5",
                "abundance.h5",
                "abundance.h5",
                "abundance.h5",
                "abundance.h5",
                "abundance.tsv",
                "abundance.tsv",
                "abundance.tsv",
                "abundance.tsv",
                "abundance.tsv",
                "bbsplit",
                "custom",
                "cutadapt_filtered_reads_plot-cnt.pdf",
                "cutadapt_filtered_reads_plot-cnt.png",
                "cutadapt_filtered_reads_plot-cnt.svg",
                "cutadapt_filtered_reads_plot-pct.pdf",
                "cutadapt_filtered_reads_plot-pct.png",
                "cutadapt_filtered_reads_plot-pct.svg",
                "cutadapt_filtered_reads_plot.txt",
                "cutadapt_trimmed_sequences_plot_3_Counts.pdf",
                "cutadapt_trimmed_sequences_plot_3_Counts.png",
                "cutadapt_trimmed_sequences_plot_3_Counts.svg",
                "cutadapt_trimmed_sequences_plot_3_Counts.txt",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.pdf",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.png",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.svg",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt",
                "fastqc",
                "fastqc-status-check-heatmap.pdf",
                "fastqc-status-check-heatmap.png",
                "fastqc-status-check-heatmap.svg",
                "fastqc-status-check-heatmap.txt",
                "fastqc_overrepresented_sequences_plot.pdf",
                "fastqc_overrepresented_sequences_plot.png",
                "fastqc_overrepresented_sequences_plot.svg",
                "fastqc_overrepresented_sequences_plot.txt",
                "fastqc_per_base_n_content_plot.pdf",
                "fastqc_per_base_n_content_plot.png",
                "fastqc_per_base_n_content_plot.svg",
                "fastqc_per_base_n_content_plot.txt",
                "fastqc_per_base_sequence_quality_plot.pdf",
                "fastqc_per_base_sequence_quality_plot.png",
                "fastqc_per_base_sequence_quality_plot.svg",
                "fastqc_per_base_sequence_quality_plot.txt",
                "fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "fastqc_per_sequence_gc_content_plot_Counts.png",
                "fastqc_per_sequence_gc_content_plot_Counts.svg",
                "fastqc_per_sequence_gc_content_plot_Counts.txt",
                "fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "fastqc_per_sequence_gc_content_plot_Percentages.png",
                "fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "fastqc_per_sequence_quality_scores_plot.pdf",
                "fastqc_per_sequence_quality_scores_plot.png",
                "fastqc_per_sequence_quality_scores_plot.svg",
                "fastqc_per_sequence_quality_scores_plot.txt",
                "fastqc_sequence_counts_plot-cnt.pdf",
                "fastqc_sequence_counts_plot-cnt.png",
                "fastqc_sequence_counts_plot-cnt.svg",
                "fastqc_sequence_counts_plot-pct.pdf",
                "fastqc_sequence_counts_plot-pct.png",
                "fastqc_sequence_counts_plot-pct.svg",
                "fastqc_sequence_counts_plot.txt",
                "fastqc_sequence_duplication_levels_plot.pdf",
                "fastqc_sequence_duplication_levels_plot.png",
                "fastqc_sequence_duplication_levels_plot.svg",
                "fastqc_sequence_duplication_levels_plot.txt",
                "fastqc_sequence_length_distribution_plot.pdf",
                "fastqc_sequence_length_distribution_plot.png",
                "fastqc_sequence_length_distribution_plot.svg",
                "fastqc_sequence_length_distribution_plot.txt",
                "fastqc_top_overrepresented_sequences_table.pdf",
                "fastqc_top_overrepresented_sequences_table.png",
                "fastqc_top_overrepresented_sequences_table.svg",
                "fastqc_top_overrepresented_sequences_table.txt",
                "general_stats_table.pdf",
                "general_stats_table.png",
                "general_stats_table.svg",
                "genome_gfp.fasta",
                "genome_gfp.gtf",
                "kallisto",
                "kallisto.merged.gene_counts.SummarizedExperiment.rds",
                "kallisto.merged.gene_counts.tsv",
                "kallisto.merged.gene_counts_length_scaled.SummarizedExperiment.rds",
                "kallisto.merged.gene_counts_length_scaled.tsv",
                "kallisto.merged.gene_counts_scaled.SummarizedExperiment.rds",
                "kallisto.merged.gene_counts_scaled.tsv",
                "kallisto.merged.gene_lengths.tsv",
                "kallisto.merged.gene_tpm.tsv",
                "kallisto.merged.transcript_counts.SummarizedExperiment.rds",
                "kallisto.merged.transcript_counts.tsv",
                "kallisto.merged.transcript_lengths.tsv",
                "kallisto.merged.transcript_tpm.tsv",
                "kallisto_alignment-cnt.pdf",
                "kallisto_alignment-cnt.png",
                "kallisto_alignment-cnt.svg",
                "kallisto_alignment-pct.pdf",
                "kallisto_alignment-pct.png",
                "kallisto_alignment-pct.svg",
                "kallisto_alignment.txt",
                "kallisto_quant.log",
                "kallisto_quant.log",
                "kallisto_quant.log",
                "kallisto_quant.log",
                "kallisto_quant.log",
                "multiqc",
                "multiqc_citations.txt",
                "multiqc_cutadapt.txt",
                "multiqc_data.json",
                "multiqc_fastqc_fastqc_trimmed.txt",
                "multiqc_general_stats.txt",
                "multiqc_kallisto.txt",
                "multiqc_report.html",
                "multiqc_report_data",
                "multiqc_report_plots",
                "multiqc_software_versions.txt",
                "multiqc_sources.txt",
                "nf_core_rnaseq_software_mqc_versions.yml",
                "out",
                "pdf",
                "pipeline_info",
                "png",
                "run_info.json",
                "run_info.json",
                "run_info.json",
                "run_info.json",
                "run_info.json",
                "svg",
                "trim",
                "trimgalore",
                "tx2gene.tsv"
            ],
            [
                "RAP1_IAA_30M_REP1.stats.txt:md5,a87292c6f7166096166a9015c2e63dac",
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,cdb7d4cd19558158c86c9cfc17f6d9b7",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,3031e78f567f2d21278eb633cf5d4830",
                "RAP1_UNINDUCED_REP1.stats.txt:md5,781e49024150a8313779c574b58ad035",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt:md5,a4382128973b6dbcbcc9c3cc7263a6e8",
                "RAP1_UNINDUCED_REP2.stats.txt:md5,e455d0163bf2d988224b0c0092605213",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt:md5,91604d61bbc72d4c38df8331cd835ef3",
                "WT_REP1.stats.txt:md5,6eab6936804c1157d5e1c61c0562d7de",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,f99a0bef8bfad292319b3455a71342e3",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,05260ac3d28c148398d2ae2da98a1c2e",
                "WT_REP2.stats.txt:md5,f6df7d4d9a5b553209f31dec2a30f92b",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt:md5,ff437dfc6744a844c412593078fcc593",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt:md5,5cfdbef1faeb38d9b7e7767fdf19d4df",
                "abundance.tsv:md5,d781d4480dc636c8a3b6aa300ac184a2",
                "abundance.tsv:md5,6fd77021287360f5005c90807233ab23",
                "abundance.tsv:md5,6fb211dc8e4fd238ca68d21d6cc83cf5",
                "abundance.tsv:md5,b6ae4ef44f718a789f58dc6451b6eebb",
                "abundance.tsv:md5,a2c6108c384662f4da8c7b7fe8f27e02",
                "cutadapt_filtered_reads_plot.txt:md5,bf033e64e9d23bee85b6277f11c663f1",
                "cutadapt_trimmed_sequences_plot_3_Counts.txt:md5,13dfa866fd91dbb072689efe9aa83b1f",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt:md5,07145dd8dd3db654859b18eb0389046c",
                "fastqc-status-check-heatmap.txt:md5,22a03548736b88b23be6bc0c9ef1b4a6",
                "fastqc_overrepresented_sequences_plot.txt:md5,4adfeacd3a3a6c7c808f121b24e6b247",
                "fastqc_per_base_n_content_plot.txt:md5,418610c1ce119cb786ad434db75d366e",
                "fastqc_per_base_sequence_quality_plot.txt:md5,bd22e06e41c096ad4f745d40fe96a1e5",
                "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,004c60768ceb6197765154e3eaa37b7a",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,95d29060b687f745288ad1ec47750037",
                "fastqc_per_sequence_quality_scores_plot.txt:md5,0f9834cc19f76dd5c87cf8cba7435a7c",
                "fastqc_sequence_counts_plot.txt:md5,3861354bbedfbde7ca36a72994f9425c",
                "fastqc_sequence_duplication_levels_plot.txt:md5,c73407d55fc532e864fa1dc8dbc12874",
                "fastqc_sequence_length_distribution_plot.txt:md5,6fe2c985606abad947bcca99b015ae33",
                "fastqc_top_overrepresented_sequences_table.txt:md5,6ef1543c6c128d51cc2b702a96c7ef63",
                "kallisto.merged.gene_counts.tsv:md5,70b5a70a847e2469a3fb3b1ea4886966",
                "kallisto.merged.gene_counts_length_scaled.tsv:md5,9b85077c766afe71785eb015b09d5a78",
                "kallisto.merged.gene_counts_scaled.tsv:md5,85e383eff98eedbc678175c8aa873ab0",
                "kallisto.merged.gene_lengths.tsv:md5,cf249fee85862c2df5a4f76244d6f701",
                "kallisto.merged.gene_tpm.tsv:md5,baae791ff46f457056571b6923843c28",
                "kallisto.merged.transcript_counts.tsv:md5,85d11e775aab1bba3c0c5afc97c57129",
                "kallisto.merged.transcript_lengths.tsv:md5,4f13d812e9eb8b4d85d12b0721d7a37d",
                "kallisto.merged.transcript_tpm.tsv:md5,89a1d81d93c9dbebdba55e28b52d6ac2",
                "kallisto_alignment.txt:md5,fafc53710b8d3cd6d3e456a1bd8c30e0",
                "multiqc_citations.txt:md5,4cece87a056a29c1338277736855e6ee",
                "multiqc_cutadapt.txt:md5,aac9581a5670cb55edf564f3d6c1f9a7",
                "multiqc_fastqc_fastqc_trimmed.txt:md5,a3238f515e01d158d875d69968753804",
                "multiqc_general_stats.txt:md5,081a8c7890b8c68da5fcf88509b09648",
                "multiqc_kallisto.txt:md5,48b29e88fbb64db2069316c6c2f662c3",
                "run_info.json:md5,2f054b3d7acf563f7c22c006dea7f206",
                "run_info.json:md5,51bc7adda1942a208b9bdee9bf9214e4",
                "run_info.json:md5,5fa89f779d412f60cb022615bf22d93c",
                "run_info.json:md5,a475f356826703868f8d7ed21124babf",
                "run_info.json:md5,1ddaaf80c06118def7bf6f374ac10b86",
                "tx2gene.tsv:md5,0e2418a69d2eba45097ebffc2f700bfe"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T11:27:10.252249"
    },
    "Params: --pseudo_aligner kallisto --skip_qc --skip_alignment - stub": {
        "content": [
            {
                "BBMAP_BBSPLIT": {
                    "bbmap": 39.01
                },
                "CAT_FASTQ": {
                    "cat": 8.3
                },
                "CUSTOM_CATADDITIONALFASTA": {
                    "python": null
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "GTF_FILTER": {
                    "python": "3.9.5"
                },
                "GUNZIP_ADDITIONAL_FASTA": {
                    "gunzip": 1.1
                },
                "GUNZIP_GTF": {
                    "gunzip": 1.1
                },
                "KALLISTO_INDEX": {
                    "kallisto": "0.48.0"
                },
                "TRIMGALORE": {
                    "trimgalore": "0.6.7",
                    "cutadapt": 3.4
                },
                "UNTAR_SALMON_INDEX": {
                    "untar": 1.34
                },
                "Workflow": {
                    "nf-core/rnaseq": "v3.16.0dev"
                }
            },
            [
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt",
                "custom",
                "fastqc",
                "genome_transcriptome.fasta",
                "genome_transcriptome.gtf",
                "multiqc",
                "multiqc_data",
                "multiqc_plots",
                "multiqc_report.html",
                "nf_core_rnaseq_software_mqc_versions.yml",
                "out",
                "pipeline_info",
                "trim",
                "trimgalore"
            ],
            [
                "RAP1_IAA_30M_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "RAP1_IAA_30M_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "RAP1_UNINDUCED_REP1_trimmed.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "RAP1_UNINDUCED_REP2_trimmed.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP1_trimmed_1.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP1_trimmed_2.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP2_trimmed_1.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "WT_REP2_trimmed_2.fastq.gz_trimming_report.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-13T11:28:04.202471"
    }
}